### PR TITLE
Moved Hacks::AnswerExercise to lib/tasks/demo

### PR DESCRIPTION
**Pull request by Dantemss**
Also namespaced the Demo stuff.

### CODE REVIEW

File: lib/tasks/demo/demo_base.rb [hunk 1]
```diff
@@ -481,7 +481,7 @@ def work_step(step, response)
     response = (response == '0' ? false : true) if response.is_a?(String)
 
     if step.tasked.exercise?
-      Hacks::AnswerExercise.call(task_step: step, is_correct: response)
+      AnswerExercise.call(task_step: step, is_correct: response)
```
Comment (jpslav) [inline lib/tasks/demo/demo_base.rb]: How is this `require`'d now that it is in `/spec`?

Comment (jpslav) [inline lib/tasks/demo/demo_base.rb]: I'm guessing the demo spec passes because it loads `/spec/support`, but that a straight demo run would blow up?

Comment (Dantemss) [inline lib/tasks/demo/demo_base.rb]: Hmm I suppose you are right... Wonder if AnswerExercise should live in the demo stuff instead?

Comment (jpslav) [inline lib/tasks/demo/demo_base.rb]: there's always the "hacks" ss ;-)

wherever you want is fine with me.

**Pull request merged**